### PR TITLE
Add feature to import/export blocked numbers for dialer

### PR DESCRIPTION
**Pull request by KryptKode**
**Notes**
- add the feature to import/export blocked numbers for the Dialer app
- numbers are exported to a plain text file with `.txt` extension with each number separated by a comma(`,`)
- numbers can be imported from a plain text file that has each number separated by a comma(`,`)
- should address this [issue](https://github.com/SimpleMobileTools/Simple-Dialer/issues/113)

<h4 align="center">
<br>
Pre-Android 10 Screenshots
<h4 align="center">
<figure>
Pre-Android Q Export
<figcaption align = "center">Export </figcaption>
</figure>
<figure>
Pre-Android Q Import
<figcaption align = "center">Import</figcaption>
</figure>
<br>
<br>
<br>

**Post-Android 10 Screenshots**

<h4 align="center">
<figure>
Post-Android Q Export
<figcaption align = "center">Export</figcaption>
</figure>
<figure>
Post-Android Q Import
<figcaption align = "center">Import</figcaption>
</figure>
<br>

### CODE REVIEW

File: commons/src/main/kotlin/com/simplemobiletools/commons/activities/ManageBlockedNumbersActivity.kt [hunk 1]
```diff
@@ -51,6 +77,8 @@ class ManageBlockedNumbersActivity : BaseSimpleActivity(), RefreshRecyclerViewLi
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
         when (item.itemId) {
             R.id.add_blocked_number -> addOrEditBlockedNumber()
+            R.id.import_blocked_numbers -> tryImportBlockedContacts()
```
Comment (tibbi) [inline commons/src/main/kotlin/com/simplemobiletools/commons/activities/ManageBlockedNumbersActivity.kt:80]: call it tryImportBlockedNumbers for consistency

File: commons/src/main/kotlin/com/simplemobiletools/commons/activities/ManageBlockedNumbersActivity.kt [hunk 2]
```diff
@@ -87,11 +120,134 @@ class ManageBlockedNumbersActivity : BaseSimpleActivity(), RefreshRecyclerViewLi
...
+        ) {
+            val outputStream = contentResolver.openOutputStream(resultData.data!!)
+            exportBlockedNumbersTo(outputStream)
         }
     }
+
+    companion object {
+        private const val PICK_IMPORT_SOURCE_INTENT = 11
+        private const val PICK_EXPORT_FILE_INTENT = 21
+    }
```
Comment (tibbi) [inline commons/src/main/kotlin/com/simplemobiletools/commons/activities/ManageBlockedNumbersActivity.kt:252]: lets keep all constants and variables at the top

File: commons/src/main/kotlin/com/simplemobiletools/commons/dialogs/ExportBlockedNumbersDialog.kt [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                            filename.isAValidFilename() -> {
+                                val file =
+                                    File(realPath, "$filename$BLOCKED_NUMBERS_EXPORT_EXTENSION")
+                                if (!hidePath && file.exists()) {
+                                    activity.toast(R.string.name_taken)
+                                    return@setOnClickListener
+                                }
+
+                                ensureBackgroundThread {
+                                    config.lastBlockedNumbersExportPath =
```
Comment (tibbi) [inline commons/src/main/kotlin/com/simplemobiletools/commons/dialogs/ExportBlockedNumbersDialog.kt:70]: do not split such things into multiple lines unnecessarily, it isnt too long. Check other places too.

File: commons/src/main/kotlin/com/simplemobiletools/commons/dialogs/ExportBlockedNumbersDialog.kt [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+import com.simplemobiletools.commons.helpers.BLOCKED_NUMBERS_EXPORT_EXTENSION
+import com.simplemobiletools.commons.helpers.ensureBackgroundThread
+import java.io.File
+import kotlinx.android.synthetic.main.dialog_export_blocked_numbers.view.export_blocked_numbers_filename
+import kotlinx.android.synthetic.main.dialog_export_blocked_numbers.view.export_blocked_numbers_folder
+import kotlinx.android.synthetic.main.dialog_export_blocked_numbers.view.export_blocked_numbers_folder_label
+
+@SuppressLint("SetTextI18n")
+class ExportBlockedNumbersDialog(
+    val activity: BaseSimpleActivity, val path: String, val hidePath: Boolean,
```
Comment (tibbi) [inline commons/src/main/kotlin/com/simplemobiletools/commons/dialogs/ExportBlockedNumbersDialog.kt:26]: this is too confusing. Either have a new line for each argument, or put them all together. Also make sure you do Ctrl + Alt + O for optimizing imports in every file

File: commons/src/main/kotlin/com/simplemobiletools/commons/dialogs/ExportBlockedNumbersDialog.kt [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    callback: (file: File) -> Unit
+) {
+    private var realPath = if (path.isEmpty()) activity.internalStoragePath else path
+    private val config = activity.baseConfig
+
+    init {
+
+        val view =
```
Comment (tibbi) [inline commons/src/main/kotlin/com/simplemobiletools/commons/dialogs/ExportBlockedNumbersDialog.kt:34]: this shouldnt be separated into 2 lines

File: commons/src/main/kotlin/com/simplemobiletools/commons/activities/ManageBlockedNumbersActivity.kt [hunk 6]
```diff
@@ -1,23 +1,52 @@
 package com.simplemobiletools.commons.activities
 
+import android.app.Activity
```
Comment (tibbi) [inline commons/src/main/kotlin/com/simplemobiletools/commons/activities/ManageBlockedNumbersActivity.kt:3]: optimize imports with ctrl + alt + o, theres too many new imports

File: commons/src/main/kotlin/com/simplemobiletools/commons/activities/ManageBlockedNumbersActivity.kt [hunk 7]
```diff
@@ -61,15 +92,20 @@ class ManageBlockedNumbersActivity : BaseSimpleActivity(), RefreshRecyclerViewLi
     }
 
     private fun updatePlaceholderTexts() {
-        manage_blocked_numbers_placeholder.text = getString(if (isDefaultDialer()) R.string.not_blocking_anyone else R.string.must_make_default_dialer)
-        manage_blocked_numbers_placeholder_2.text = getString(if (isDefaultDialer()) R.string.add_a_blocked_number else R.string.set_as_default)
+        manage_blocked_numbers_placeholder.text =
+            getString(if (isDefaultDialer()) R.string.not_blocking_anyone else R.string.must_make_default_dialer)
+        manage_blocked_numbers_placeholder_2.text =
+            getString(if (isDefaultDialer()) R.string.add_a_blocked_number else R.string.set_as_default)
```
Comment (tibbi) [inline commons/src/main/kotlin/com/simplemobiletools/commons/activities/ManageBlockedNumbersActivity.kt:98]: splitting these lines wasnt needed, they dont go past the 160 limit

File: commons/src/main/kotlin/com/simplemobiletools/commons/dialogs/ExportBlockedNumbersDialog.kt [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+            .setPositiveButton(R.string.ok, null)
+            .setNegativeButton(R.string.cancel, null)
+            .create().apply {
+                activity.setupDialogStuff(view, this, R.string.export_blocked_numbers) {
+                    getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener {
+                        val filename = view.export_blocked_numbers_filename.value
+                        when {
+                            filename.isEmpty() -> activity.toast(R.string.empty_name)
+                            filename.isAValidFilename() -> {
+                                val file =
```
Comment (tibbi) [inline commons/src/main/kotlin/com/simplemobiletools/commons/dialogs/ExportBlockedNumbersDialog.kt:62]: do not split such lines either

File: commons/src/main/kotlin/com/simplemobiletools/commons/activities/ManageBlockedNumbersActivity.kt (hunk 6 above)
Comment (KryptKode) [reply commons/src/main/kotlin/com/simplemobiletools/commons/activities/ManageBlockedNumbersActivity.kt:3]: What's your take on the starred imports?

Comment (KryptKode) [reply commons/src/main/kotlin/com/simplemobiletools/commons/activities/ManageBlockedNumbersActivity.kt:3]: I was using single name imports that's why. Do we need to add a limit in the .editorconfig? Maybe something like this
```
[*.kt]
ij_kotlin_name_count_to_use_star_import = 5
ij_kotlin_name_count_to_use_star_import_for_members = 5
```

Comment (tibbi) [reply commons/src/main/kotlin/com/simplemobiletools/commons/activities/ManageBlockedNumbersActivity.kt:3]: I use the default values everywhere to keep it simple. Anyway, added the new rules into editorconfig.

**Pull request merged**